utils/swift_build_support/swift_build_support/products/wasmstdlib.py

```diff
@@ -194,3 +197,94 @@ def add_extra_cmake_options(self):
                                   '-Xcc;-mthread-model;-Xcc;posix;'
                                   '-Xcc;-pthread;-Xcc;-ftls-model=local-exec')
         self.cmake_options.define('SWIFT_ENABLE_WASI_THREADS:BOOL', 'TRUE')
+
+
+class WasmSwiftSDK(product.Product):
+    @classmethod
+    def product_source_name(cls):
+        return "swift-sdk-generator"
+
+    @classmethod
+    def is_build_script_impl_product(cls):
+        return False
+
+    @classmethod
+    def is_before_build_script_impl_product(cls):
+        return False
+
+    def should_build(self, host_target):
+        return self.args.build_wasmstdlib
+
+    def should_test(self, host_target):
+        return False
+
+    def _target_package_path(self, target_triple):
+        return os.path.join(self.build_dir, 'Toolchains', target_triple)
+
+    def _build_target_package(self, target_triple,
+                              stdlib_build_path, llvm_runtime_libs_build_path):
+
+        dest_dir = self._target_package_path(target_triple)
+        shell.rmtree(dest_dir)
+        shell.makedirs(dest_dir)
+
+        # Build toolchain package for standalone stdlib
+        with shell.pushd(stdlib_build_path):
+            shell.call([self.toolchain.cmake, '--install', '.'],
+                       env={'DESTDIR': dest_dir})
+
+        # Copy LLVM runtime libraries
+        with shell.pushd(llvm_runtime_libs_build_path):
+            clang_dest_dir = os.path.join(dest_dir, 'usr/lib/swift_static/clang')
+            shell.call([self.toolchain.cmake, '--install', '.',
+                        '--component', 'clang_rt.builtins-wasm32'],
+                       env={'DESTDIR': clang_dest_dir})
+        return dest_dir
+
+    def build(self, host_target):
+        build_root = os.path.dirname(self.build_dir)
+        llvm_runtime_libs_build_path = os.path.join(
+            build_root, '%s-%s' % ('wasmllvmruntimelibs', host_target))
+
+        target_packages = []
+        for target_triple, build_basename in [
+            ('wasm32-unknown-wasi', 'wasmstdlib'),
+            # TODO: Enable threads stdlib once sdk-generator supports multi-target SDK
+            # ('wasm32-unknown-wasip1-threads', 'wasmthreadsstdlib'),
+        ]:
+            stdlib_build_path = os.path.join(
+                build_root, '%s-%s' % (build_basename, host_target))
+            package_path = self._build_target_package(
+                target_triple, stdlib_build_path, llvm_runtime_libs_build_path)
+            target_packages.append((target_triple, package_path))
+
+        swiftc_path = os.path.abspath(self.toolchain.swiftc)
```

Comment:
Why are you using the prebuilt 5.8 toolchain on the CI after building a fresh trunk 6.1 toolchain? Most build products use [that freshly built trunk `native_toolchain_path()`](https://github.com/apple/swift/blob/ae6ff02d1be440faa6e4e14296b04fbb02d341fb/utils/swift_build_support/swift_build_support/products/swiftpm.py#L49).

Reply:
The Swift SDK will be used to test SwiftPM, so it should be built before SwiftPM and we cannot use just built SwiftPM here.

Reply:
OK, you are currently building this Wasm SDK at the end though, after a bunch of SwiftPM products are also built with trunk SwiftPM. You will have to move the Wasm SDK build much earlier to use it to test SwiftPM.

Reply:
Right, the current position in the pipeline is too late.